### PR TITLE
 BOAC-440 Initialize/drop BOAC DB schema from SQL file

### PR DESCRIPTION
**Pull request by raydavis**
https://jira.ets.berkeley.edu/jira/browse/BOAC-440

We intended to do this before the first production release, but what's a minor version among friends?

### CODE REVIEW

File: boac/models/alert.py [hunk 1]
```diff
@@ -95,7 +95,7 @@ def result_to_dict(result):
     @classmethod
     def current_alerts_for_sid(cls, viewer_id, sid):
         query = AlertView.query.filter_by(viewer_id=viewer_id, dismissed_at=None).join(cls).filter_by(sid=sid, active=True)
-        results = query.all()
+        results = query.order_by(Alert.created_at).all()
```
Comment (raydavis) [inline boac/models/alert.py]: `test_alerts_controller.py` assumes a certain order and was failing semi-randomly as a result.

File: boac/models/development_db.py [hunk 2]
```diff
@@ -66,7 +67,10 @@
 
 
 def clear():
-    db.drop_all()
+    with open('scripts/db/drop_schema.sql', 'r') as ddlfile:
```
Comment (johncrossman) [inline boac/models/development_db.py]: @raydavis I assume the above is fine but absolute path (using `app.config['BASE_DIR']`) seems to be an unofficial (boac) convention.

Comment (raydavis) [reply boac/models/development_db.py]: I'll amend.

File: scripts/db/drop_schema.sql [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+ALTER TABLE IF EXISTS ONLY public.athletics DROP CONSTRAINT IF EXISTS athletics_pkey;
+ALTER TABLE IF EXISTS ONLY public.alerts DROP CONSTRAINT IF EXISTS alerts_sid_alert_type_key_unique_constraint;
+ALTER TABLE IF EXISTS ONLY public.alerts DROP CONSTRAINT IF EXISTS alerts_pkey;
+ALTER TABLE IF EXISTS ONLY public.alert_views DROP CONSTRAINT IF EXISTS alert_views_pkey;
+ALTER TABLE IF EXISTS ONLY public.alembic_version DROP CONSTRAINT IF EXISTS alembic_version_pkc;
+ALTER TABLE IF EXISTS ONLY public.advisor_watchlists DROP CONSTRAINT IF EXISTS advisor_watchlists_pkey;
+ALTER TABLE IF EXISTS public.json_cache ALTER COLUMN id DROP DEFAULT;
+ALTER TABLE IF EXISTS public.cohort_filters ALTER COLUMN id DROP DEFAULT;
+ALTER TABLE IF EXISTS public.authorized_users ALTER COLUMN id DROP DEFAULT;
+ALTER TABLE IF EXISTS public.alerts ALTER COLUMN id DROP DEFAULT;
```
Comment (johncrossman) [inline scripts/db/drop_schema.sql]: @raydavis much of the above is due to our missing `ON DELETE CASCADE`?  Want me to create a Jira to add 'delete cascade'?  We can tackle later.

Comment (raydavis) [reply scripts/db/drop_schema.sql]: @johncrossman The drop_schema.sql file is meant to replace the old ORM code in `development_db.clear()`. It's only used by the test suite, so as to get a completely fresh test DB. This SQL was obligingly supplied by 
```
pg_dump --clean --if-exists --schema-only -U boac postgresql://.../boac_shared > boac_shared_schema.sql
```

The delete cascade migrations will be in a migrate script, which I'm working on next.

**Comment by johncrossman** [comment]
Looks good, @raydavis  When green, I'll merge.

**Pull request merged**